### PR TITLE
EnumConverter: Added ability to write enum as number

### PR DESCRIPTION
**Pull request by jformacek**
Hello,
proposing one more change to project that came handy in my recent project: extension of EnumConverter that allows to write enum values not only as strings, but also as integer values.

Thanks for maintaining this project,
Jiri

### CODE REVIEW

File: FileHelpers/Converters/EnumConverter.cs [hunk 1]
```diff
@@ -7,23 +7,76 @@ namespace FileHelpers
...
+            : this(sourceEnum, "s")
+        { }
+           
+
+        public EnumConverter(Type sourceEnum, string format)
         {
             if (sourceEnum.IsEnum == false)
                 throw new BadUsageException("The Input sourceType must be an Enum but is of type " + sourceEnum.Name);
 
             mEnumType = sourceEnum;
```
Comment (mcavigelli) [inline FileHelpers/Converters/EnumConverter.cs]: You came up with very simple solution, great!

I suggest to simplify the switch case statement by "lowering" the format string. Then one needs just to one case. Maybe you could also extract a method doing the parsing:

``` cs
 private static EnumFormat GetEnumFormat(string format)
 {
       switch (format.Trim().ToLower())
       {
            case "s":
                return EnumFormat.String;
             case "n":
                return EnumFormat.Number;
            default:
                throw new BadUsageException("The format parameter must be either \"s\" (converts enum to string) or \"n\" (converts enum to number).");
       }
}
```

File: FileHelpers/Converters/EnumConverter.cs [hunk 2]
```diff
@@ -7,23 +7,76 @@ namespace FileHelpers
...
+
+            /// <summary>
+            /// as integer
+            /// </summary>
+            Number,
+        }
 
         public EnumConverter(Type sourceEnum)
+
+            : this(sourceEnum, "s")
```
Comment (mcavigelli) [inline FileHelpers/Converters/EnumConverter.cs]: I like the idea of chaining constructors. However it looks like a magic string. "s" and "n" should just be defined once in the code.

My proposal is that you change the visibility of ```EnumFormat``` to ```internal```.
The two constructors would have the following signatures:

```cs
public EnumConverter(Type sourceEnum, EnumFormat format = EnumFormat.String) {}

public EnumConverter(Type sourceEnum, string format) : this(sourceEnum, GetEnumFormat(format))
{
}

```

File: FileHelpers/Converters/EnumConverter.cs [hunk 3]
```diff
@@ -7,23 +7,76 @@ namespace FileHelpers
...
+                default:
+                    throw new BadUsageException("The format parameter must be either \"s\" (converts enum to string) or \"n\" (converts enum to number).");
+
+            }
         }
 
         public override object StringToField(string from)
         {
-            try {
+            try
```
Comment (jformacek) [inline FileHelpers/Converters/EnumConverter.cs]: Hello Matthias,
thanks for feedback.
RE formatting of try/catch block: this is something editor has done for me along the way and I haven't noticed...

RE Tests: I added tests for writing enum type in both formats - thanks for asking for this.

RE Examples: I added example of usage of format string when working with enums; in separate commit I also added example of usage of engine.NewLineForWriting that I added recently

Thank you for working on this project,
Jiri

File: FileHelpers/Converters/EnumConverter.cs (hunk 2 above)
Comment (jformacek) [inline FileHelpers/Converters/EnumConverter.cs]: Updated as proposed - thanks!

File: FileHelpers/Converters/EnumConverter.cs (hunk 1 above)
Comment (jformacek) [inline FileHelpers/Converters/EnumConverter.cs]: Updated as proposed - thanks!

**Comment by mcavigelli** [comment]
Hi Jiri

Thank you for your extension for FileHelpers!
Thanks for the tests which help us prevent regressions.
And finally thank you for even having extended the examples.

Matthias

**Comment by jformacek** [comment]
Hi Matthias, I there planned release sometimes soon? I now use Filehelpers with my customizations, and would be great to switch to an "official" release. 
Thanks,
Jiri

**Comment by guillaumejay** [comment]
An updated package would be great.

(It would be nice to have annotation on properties #67)

**Comment by mcavigelli** [comment]
Hi Jiri

I'm also looking forward to make a new release / nuget package.
It would be nice to make appveyor with every build a alpha / beta nuget package.
The build scripts are around, but not connected with appveyor yet.

Matthias

**Pull request merged**